src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/InMemoryEventStore.java

```diff
@@ -92,17 +92,17 @@ public BatchPayload getEventsBatch(int numberToGet) {
      * the events are deleted from the InMemoryEventStore. If not, they are reinserted at the beginning
      * of the buffer queue for another attempt.
      *
-     * @param successfullySent if the batch of events was successfully sent
+     * @param delete if false, move events back to the buffer instead of deleting
      * @param batchId the ID of the batch of events
      */
     @Override
-    public void cleanupAfterSendingAttempt(boolean successfullySent, long batchId) {
+    public void deleteBatchedEvents(boolean delete, long batchId) {
         // Events that successfully sent are deleted from the pending buffer
         List<TrackerPayload> events = eventsBeingSent.remove(batchId);
 
         // Events that didn't send are inserted at the head of the eventBuffer
         // for immediate resending.
-        if (!successfullySent) {
+        if (!delete) {
```

Comment:
Nitpicky... but I liked the previous name as it was less dependent by the actual implementation IMO. If we want to use the new one I think the `delete` param is a bit cryptic, as the method name already says _delete_. Would something like `needRetry` work? (the logic would be inverted in this case).